sympy/solvers/solvers.py

```diff
@@ -228,7 +228,7 @@ def checksol(f, symbol, sol=None, **flags):
                     return False
                 # there are free symbols -- simple expansion might work
                 _, val = val.as_content_primitive()
-                val = expand_mul(expand_multinomial(val))
+                val = simplify(expand_mul(expand_multinomial(val)))
```

Comment:
Then how about using simplify on val for for `attempt==2`, just after the `minimal` check? You aren't running into cases where attempting to simplify is really slow, but there are such cases and we should allow people to look for trivial solutions if they so desire. At least that is the intent of the current function.